### PR TITLE
deps: bump SDK to v0.47.1

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

Waiting on this [PR](https://github.com/strangelove-ventures/interchaintest/pull/498) in interchaintest to be merged.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
deps: bump SDK to v0.47.1
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by crodriguezvega** [comment]
I am a bit stuck here... I bump interchaintest to the latest (`v7.0.0-20230417164426-9b8f7ae9bf1d`) but I still get this error when I do `go mod tidy`:

```
github.com/cosmos/ibc-go/e2e/testconfig imports
        github.com/cometbft/cometbft/types imports
        github.com/cometbft/cometbft/crypto/secp256k1 imports
        github.com/btcsuite/btcd/btcec/v2/ecdsa tested by
        github.com/btcsuite/btcd/btcec/v2/ecdsa.test imports
        github.com/btcsuite/btcd/chaincfg/chainhash: ambiguous import: found package github.com/btcsuite/btcd/chaincfg/chainhash in multiple modules:
        github.com/btcsuite/btcd v0.22.0-beta (/Users/carlosrodriguez/go/pkg/mod/github.com/btcsuite/btcd@v0.22.0-beta/chaincfg/chainhash)
        github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1 (/Users/carlosrodriguez/go/pkg/mod/github.com/btcsuite/btcd/chaincfg/chainhash@v1.0.1)
```

interchaintest was having the same issue, but in the latest commit that was fixed and it works now. But I am lost why we have this error in ibc-go...

**Comment by crodriguezvega** [comment]
I think the problem is fixed! Thank to @misko9 for the solution to the puzzle!

**Comment by colin-axner** [comment]
looks like tests are still failing

**Comment by crodriguezvega** [comment]
> looks like tests are still failing

I know :) This is now a different problem.

**Comment by chatton** [comment]
Looking a bit more into the failures, I attempted running the tests using a simd from main, but with interchain test pointing at sdk 0.47.1 and ran into the same account sequence mismatch error. Looking at [the SDK changes](https://github.com/cosmos/cosmos-sdk/compare/v0.47.0...v0.47.1) I do suspect it has something to do with either CLI args (used by interchain test) or the `clientContext` which is also used in interchaintest is the `BroadcastTx` function which calls the sdk `BroadcastTx`.

**Comment by chatton** [comment]
After some more debugging with @charleenfei , we managed to figure it out. The problem was because of the changes in [this PR](https://github.com/cosmos/cosmos-sdk/pull/15123).

Specifically 

```diff
	initNum, initSeq := fc.accountNumber, fc.sequence
-	if initNum == 0 || initSeq == 0 {
+     if initNum == 0 && initSeq == 0 {
		num, seq, err := fc.accountRetriever.GetAccountNumberSequence(clientCtx, from)
		if err != nil {
			return fc, err
		}

-		if initNum == 0 {
-			fc = fc.WithAccountNumber(num)
-		}
+ fc = fc.WithAccountNumber(num)
+ fc = fc.WithSequence(seq)
-
-		if initSeq == 0 {
-			fc = fc.WithSequence(seq)
-		}
	}
```

In interchain test, we were accidentally not explicitly passing the account sequence when creating the [tx Factory](https://github.com/strangelove-ventures/interchaintest/blob/a06599d2eb8c9c59e70158ae0339cd7b5ab6377a/chain/cosmos/broadcaster.go#L169-L183). As a result, the `||` conditional in the above code made it so that we correctly got the sequence. 

I will create a follow up PR in interchain test to fix this issue there.

I believe there should be a check in the sdk that ensures if exactly one of the account / account sequence is zero, it should error out.

File: e2e/testsuite/tx.go [hunk 1]
```diff
@@ -31,15 +31,29 @@ import (
...
+			panic(err)
+		}
+
 		cdc := Codec()
 		return clientContext.WithCodec(cdc).WithTxConfig(authtx.NewTxConfig(cdc, []signingtypes.SignMode{signingtypes.SignMode_SIGN_MODE_DIRECT}))
 	})
 
 	broadcaster.ConfigureFactoryOptions(func(factory tx.Factory) tx.Factory {
-		return factory.WithGas(DefaultGasValue)
+		return factory.WithGas(DefaultGasValue).WithSequence(seq)
```
Comment (charleenfei) [inline e2e/testsuite/tx.go:56]: 🎉

File: e2e/testsuite/tx.go [hunk 2]
```diff
@@ -31,15 +31,29 @@ import (
...
+		sdkAdd, err := sdk.AccAddressFromBech32(user.FormattedAddress())
+		if err != nil {
+			panic(err)
+		}
+
+		_, seq, err = clientContext.AccountRetriever.GetAccountNumberSequence(clientContext, sdkAdd)
+		if err != nil {
+			panic(err)
+		}
+
```
Comment (chatton) [inline e2e/testsuite/tx.go:50]: this can be removed when [this issue](https://github.com/strangelove-ventures/interchaintest/issues/530) is resolved.

**Review by DimitrisJim — APPROVED** [review]
dope working on debugging the issue @chatton & @charleenfei

**Review by colin-axner — APPROVED** [review]

**Comment by colin-axner** [comment]
Fantastic work y'all!

**Pull request merged**